simplified-ui-splash/src/main/java/org/nypl/simplified/ui/splash/MailtoWebViewClient.java

```diff
@@ -0,0 +1,48 @@
+package org.nypl.simplified.ui.splash;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.MailTo;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import java.lang.ref.WeakReference;
+
+/**
+ * Used to handle mailto: links in the eula.html
+ */
+public class MailtoWebViewClient extends WebViewClient {
+    private final WeakReference<Activity> mActivityRef;
+
+    public MailtoWebViewClient(Activity activity) {
+        mActivityRef = new WeakReference<>(activity);
+    }
+
+    @Override
+    public boolean shouldOverrideUrlLoading(WebView view, String url) {
+        if (url.startsWith("mailto:")) {
+            final Activity activity = mActivityRef.get();
+            if (activity != null) {
+                MailTo mt = MailTo.parse(url);
+                Intent i = newEmailIntent(activity, mt.getTo(), mt.getSubject(), mt.getBody(), mt.getCc());
+                activity.startActivity(i);
+                view.reload();
+                return true;
+            }
+        } else {
+            view.loadUrl(url);
+        }
+        return true;
+    }
+
+    private Intent newEmailIntent(Context context, String address, String subject, String body, String cc) {
+        Intent intent = new Intent(Intent.ACTION_SEND);
+        intent.putExtra(Intent.EXTRA_EMAIL, new String[] { address });
+        intent.putExtra(Intent.EXTRA_TEXT, body);
+        intent.putExtra(Intent.EXTRA_SUBJECT, subject);
+        intent.putExtra(Intent.EXTRA_CC, cc);
+        intent.setType("message/rfc822");
```

Comment:
This is a bit of a hack (I found the StackOverflow that suggested this).

It's always been a bit of black magic to figure out the right combination of Intent extras to compose an email.

What you really want here is `Intent.ACTION_SENDTO` instead of the more generic `Intent.ACTION_SEND`. Try something like this:

```java
    Intent intent = new Intent(Intent.ACTION_SENDTO);
    intent.setData(Uri.parse("mailto:")); // only email apps should handle this
    intent.putExtra(Intent.EXTRA_EMAIL, addresses);
    intent.putExtra(Intent.EXTRA_SUBJECT, subject);
    intent.putExtra(Intent.EXTRA_CC, cc);
    if (intent.resolveActivity(getPackageManager()) != null) {
        startActivity(intent);
    }
```

As a bonus this accounts for the lack of a Mail app as well.

https://developer.android.com/guide/components/intents-common#ComposeEmail